File: app/src/main/java/vassar/cmpu203/vassardiningapp/view/RestrictionIconAdapter.java
package vassar.cmpu203.vassardiningapp.view;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.Set;

import vassar.cmpu203.vassardiningapp.databinding.RestrictionIconBinding;
import vassar.cmpu203.vassardiningapp.model.DietaryRestriction;

public class RestrictionIconAdapter extends RecyclerView.Adapter<RestrictionIconAdapter.ViewHolder> {

    private final Set<DietaryRestriction> restrictions;

    public RestrictionIconAdapter(Set<DietaryRestriction> restrictions) {
        this.restrictions = restrictions;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(RestrictionIconBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.setData(new ArrayList<>(restrictions).get(position));
    }

    @Override
    public int getItemCount() {
        return restrictions.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        private final RestrictionIconBinding binding;

        public ViewHolder(@NonNull RestrictionIconBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void setData(DietaryRestriction restriction) {
            binding.restrictionIcon.setImageResource(restriction.getIconId());
        }
    }
}

File: androidApp/src/main/java/vassar/cmpu203/vassardiningapp/view/RestrictionIconAdapter.kt
package vassar.cmpu203.vassardiningapp.view

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import vassar.cmpu203.vassardiningapp.databinding.RestrictionIconBinding
import com.vassar.vassardiningappcommon.DietaryRestriction
import vassar.cmpu203.vassardiningapp.R

class RestrictionIconAdapter(private val restrictions: Set<DietaryRestriction>) :
        RecyclerView.Adapter<RestrictionIconAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(RestrictionIconBinding.inflate(LayoutInflater.from(parent.context), parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.setData(ArrayList(restrictions)[position])
    }

    override fun getItemCount(): Int = restrictions.size

    class ViewHolder(private val binding: RestrictionIconBinding) : RecyclerView.ViewHolder(binding.root) {
        fun setData(restriction: DietaryRestriction) {
            binding.restrictionIcon.setImageResource(binding.root.context.resources.getIdentifier(restriction.iconId, "drawable", binding.root.context.packageName))
        }
    }
}
